Core/UserAgentManager.swift

```diff
@@ -253,32 +253,44 @@ struct UserAgent {
     private func closestLogic(forUrl url: URL?,
                               isDesktop: Bool,
                               privacyConfig: PrivacyConfiguration) -> String {
-        guard shouldUseUpdateLogic else {
+        guard shouldUseUpdatedLogic else {
             return oldLogic(forUrl: url, isDesktop: isDesktop, privacyConfig: privacyConfig)
         }
 
         if isDesktop {
             return "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.5 Safari/605.1.15"
         }
-        var deviceProfile = "iPhone; CPU iPhone OS 16_6 like Mac OS X"
-        if baseAgent.contains("iPad") {
-            deviceProfile = "iPad; CPU OS 16_6 like Mac OS X"
-        } else if baseAgent.contains("iPod") {
-            deviceProfile = "iPod touch; CPU iPhone 16_6 like Mac OS X"
-        }
+
         return "Mozilla/5.0 (" + deviceProfile + ") AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.5 Mobile/15E148 Safari/604.1"
     }
 
-    private var shouldUseUpdateLogic: Bool {
-        // Unfortunately you can't combine #available with any other condition in the same line
-        if isTesting {
-            return true
-        } else if #available(iOS 16.6, *) {
-            return true
+    private var shouldUseUpdatedLogic: Bool {
+        guard let webKitVersion else { return false }
+        return webKitVersion.versionCompare("605.1.15") != .orderedAscending
```

Comment:
I think this will be true for "605.1.14" and "605.1.16" etc which is incorrect.

I think the following might be what we're wanting:
```suggestion
        return webKitVersion.versionCompare("605.1.14") == .orderedAscending
```

Reply:
I believe this is fine - added two tests that should validate that - please could you take a look at them?